### PR TITLE
BufferAttribute: Deprecate .dynamic, introduce .usage.

### LINKED ISSUE #14730: BufferAttribute.dynamic: revisited
This discussion also applies to `InterleavedBuffer.dynamic`.

It seems that the `dynamic` property is being used for multiple purposes:

1. as a _hint_ as to how frequently the data will be updated, [[docs](https://github.com/mrdoob/three.js/blob/master/docs/api/core/BufferAttribute.html#L61-L69)], and

2. as a flag, that when `true`, allows the updating of only a sub-rage, as specified in the `updateRange` property. [[code](https://github.com/mrdoob/three.js/blob/master/src/renderers/webgl/WebGLAttributes.js#L66-L96)].

The examples are not consistent here. Even in examples in which the entire range is being updated each frame, sometimes the dynamic property is set to `true`, and sometimes it is not.

This has been discussed [before](https://github.com/mrdoob/three.js/pull/9631#issuecomment-254375571), and the purpose of the property was modified at that time.

What should we advise users to do in setting this property?

/ping @benaadams 
/ping @mattdesl 
/ping @mrdoob 

- [ x ] Dev
- [ x ] r95

### PR DESCRIPTION
**Pull request by Mugen87**
Fixed #14730

I've managed it to make the change backwards-compatible. So `BufferAttribute.dynamic` and `BufferAttribute.setDynamic()` can still be used but with warnings. Same for `InterleavedBuffer`'s interface.

The new `usage` property directly maps to the same-named parameter of [WebGLRenderingContext.bufferData()](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferData). So this property is now only intended to define the usage pattern of the buffer data for optimization purposes.

`WebGLAttributes.updateBuffer()` only uses `bufferSubData()` now. Keep in mind that the initial buffer upload is done with `bufferData()` as before.

### CODE REVIEW

**Comment by Mugen87** [comment]
Merging see https://github.com/mrdoob/three.js/pull/17454#issuecomment-531628734

The sooner it lands in `dev`, the more time we have for verification.

**Comment by Mugen87** [comment]
Updated builds: 3a3d2b1e591ced02b5742c6f4dd5d62f457a760c

File: docs/api/en/core/BufferAttribute.html [hunk 1]
```diff
@@ -178,8 +174,8 @@ <h3>[method:BufferAttribute set] ( [param:Array value], [param:Integer offset] )
 		being a [page:TypedArray].
 		</p>
 
-		<h3>[method:BufferAttribute setDynamic] ( [param:Boolean value] ) </h3>
-		<p>Set [page:BufferAttribute.dynamic dynamic] to value.</p>
+		<h3>[method:BufferAttribute setUsage] ( [param:Usage value] ) </h3>
+		<p>Set [page:BufferAttribute.usage usage] to value.</p>
```
Comment (WestLangley) [inline docs/api/en/core/BufferAttribute.html]: But what is it? I think you should describe what it is.

**Comment by mrdoob** [comment]
Thanks!

File: docs/api/en/core/BufferAttribute.html [hunk 2]
```diff
@@ -111,10 +101,16 @@ <h3>[property:Object updateRange]</h3>
 			related to color).
 		</p>
 
+		<h3>[property:Usage usage]</h3>
+		<p>
+			Defines the intended usage pattern of the data store for optimization purposes. Corresponds to the *usage* parameter of
+			[link:https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferData WebGLRenderingContext.bufferData]().
+			Default is *THREE.StaticDrawUsage*.
```
Comment (WestLangley) [inline docs/api/en/core/BufferAttribute.html]: This needs to link to a description of the values it can take.

**Pull request merged**